File: Java/Practice/LeetCode/Medium/Maximum Number of Coins You Can Get.java
// https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/

class Solution {
    public int maxCoins(int[] piles) {
        int sum = 0;
        int counter = 0;
        int idx = piles.length - 2;

        Arrays.sort(piles);

        while (counter < piles.length / 3) {
            sum += piles[idx];
            idx -= 2;
            counter++;
        }

        return sum;
    }
}

File: Python/Practice/LeetCode/Medium/Maximum Number of Coins You Can Get.py
# https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/

class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        sum_val = 0
        counter = 0
        idx = 1

        piles.sort(reverse=True)

        while counter < len(piles) / 3:
            sum_val += piles[idx]
            idx += 2
            counter += 1

        return sum_val